plugins/org.polarsys.capella.cybersecurity.model.cdo/src/org/polarsys/capella/cybersecurity/model/impl/FunctionalPrimaryAssetImpl.java

```diff
@@ -153,7 +163,7 @@ public boolean eIsSet(int featureID) {
     case CybersecurityPackage.FUNCTIONAL_PRIMARY_ASSET__FUNCTIONS:
       return !getFunctions().isEmpty();
     case CybersecurityPackage.FUNCTIONAL_PRIMARY_ASSET__FUNCTIONAL_CHAINS:
-      return !getFunctionalChains().isEmpty();
+      return functionalChains != null && !functionalChains.isEmpty();

```

Comment:
Are you sure this work?
functionalChains doesn't seems to be filled somewhere?

Reply:
Yes, it seems to be working.
Took it from here (not the CDO model):
https://github.com/eclipse/capella-cybersecurity/blob/master/plugins/org.polarsys.capella.cybersecurity.model/src/org/polarsys/capella/cybersecurity/model/impl/FunctionalPrimaryAssetImpl.java